src/python/pants/backend/python/goals/export.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from dataclasses import dataclass
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import VenvPex, VenvPexProcess
+from pants.backend.python.util_rules.pex_environment import PexEnvironment
+from pants.backend.python.util_rules.pex_from_targets import PexFromTargetsRequest
+from pants.core.goals.export import ExportableData, ExportableDataRequest, ExportError, Symlink
+from pants.engine.internals.selectors import Get
+from pants.engine.process import ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+
+
+@dataclass(frozen=True)
+class ExportedVenvRequest(ExportableDataRequest):
+    pass
+
+
+@rule
+async def export_venv(
+    request: ExportedVenvRequest, python_setup: PythonSetup, pex_env: PexEnvironment
+) -> ExportableData:
+    # Pick a single interpreter for the venv.
+    interpreter_constraints = InterpreterConstraints.create_from_targets(
+        request.targets, python_setup
+    )
+    min_interpreter = interpreter_constraints.snap_to_minimum(python_setup.interpreter_universe)
+    if not min_interpreter:
+        raise ExportError(
+            "There is no single Python interpreter compatible with all the "
+            "targets for which export was requested. Please restrict the target set "
+            "to one that shares a compatible interpreter."
```

Comment:
It would probably be helpful to include the value of `interpreter_constraints` in the message.

Reply:
Done.